src/ScriptEditor/NetscriptDefinitions.d.ts

```diff
@@ -1611,7 +1611,7 @@ export interface StudyTask {
   type: "CLASS";
   cyclesWorked: number;
   classType: string;
-  location: string;
+  location: LocationName | `${LocationName}`;
```

Comment:
Did your search-and-replace go bad, or is this intentional?

Reply:
In `SleeveClassTask`, I see this:
```typescript
type SleeveClassTask = {
  type: "CLASS";
  classType: UniversityClassType | GymType | `${UniversityClassType}` | `${GymType}`;
  location: LocationName | `${LocationName}`;
};
```
Many `XXXTask` (`SleeveClassTask`, `SleeveCrimeTask`, `SleeveFactionTask`) use `property: Type | ${Type}` format, so I imitated them for `StudyTask`. I'm not sure why they are like that, though. It's pretty weird because `CompanyWorkTask`, `CrimeTask`, `FactionWorkTask` use `property: Type`.

I agree that `property: Type` looks like the proper way here. What's your suggestion?

Reply:
OK, I finally figured out what's going on here, but I'm not 100% certain of the answer either. 

@Snarling added this syntax ~two years ago, during a big refactor to add more enum use. The actual syntax is "template literal types", a TS feature that (IMO) is rather obscure, and allows you to make complicated string unions. In this case, it is making a union of all the enum values. 

Where this gets weird is that the enum type itself is *already* a union of all the enum values. However, it is a union of them *as enum value types*, rather than as strings.

Further complicating this is that *the type definitions for these functions don't matter*. Internally, all the arguments are changed to `unknown`, and we (have to) do runtime type checking anyway. So (for the functions), the stuff in NetscriptDefinitions is only for documentation. 

Given all that, I think the right course is to rip out the confusing and semi-redundant template literal types. But that's a different refactor; I went ahead and merged this.

Reply:
I'll take note of this problem and make a PR for it later.